spec/index.bs

```diff
@@ -2685,68 +2756,54 @@ A {{GPUBindGroupLayout}} object has the following internal slots:
-                                    or {{GPUBindingType/"writeonly-storage-texture"}}:
-                                        - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/viewDimension}} is `undefined`.
-
-                                - If |bindingDescriptor|.{{GPUBindGroupLayoutEntry/type}} is **not**
-                                    {{GPUBindingType/"sampled-texture"}} or {{GPUBindingType/"multisampled-texture"}}:
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/textureComponentType}} is `undefined.`
-
-                                - If |bindingDescriptor|.{{GPUBindGroupLayoutEntry/type}} is **not**
-                                    {{GPUBindingType/"readonly-storage-texture"}} or
-                                    {{GPUBindingType/"writeonly-storage-texture"}}:
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/storageTextureFormat}} is `undefined`.
-
-                                - If |bindingDescriptor|.{{GPUBindGroupLayoutEntry/type}} is {{GPUBindingType/"multisampled-texture"}}:
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/viewDimension}} is
-                                        {{GPUTextureViewDimension/2d}}.
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/textureComponentType}} is not
-                                        {{GPUTextureComponentType/depth-comparison}}.
-
-                                - If |bindingDescriptor|.{{GPUBindGroupLayoutEntry/type}} is
-                                    {{GPUBindingType/"readonly-storage-texture"}} or
-                                    {{GPUBindingType/"writeonly-storage-texture"}}:
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/viewDimension}} is not
-                                        {{GPUTextureViewDimension/cube}} or {{GPUTextureViewDimension/cube-array}}.
-                                    - |bindingDescriptor|.{{GPUBindGroupLayoutEntry/storageTextureFormat}} must
-                                        be a format which can support storage usage.
+                                    - The [$layout entry binding type$] of |bindingDescriptor| is not
+                                        {{GPUBufferType/"readonly-storage"}} or {{GPUStorageTextureAccess/"writeonly"}}.
+
+                                - If the |textureLayout| is not `undefined` and
+                                    |textureLayout|.{{GPUTextureBindingLayout/multisampled}} is `true`:
+                                    - |textureLayout|.{{GPUTextureBindingLayout/viewDimension}} is
+                                        {{GPUTextureViewDimension/"2d"}}.
+                                    - |textureLayout|.{{GPUTextureBindingLayout/type}} is not
+                                        {{GPUTextureSampleType/"float"}}.
```

Comment:
Rereading this after removing "depth" from this (sorry I didn't realize this earlier), I think this refactor had more consequences than we realized. Shouldn't it be totally fine to create a multisampled layout entry with any "float" as well? Technically anything you could do with `multisampled:true,type:"float"` you could do with `multisampled:true,type:"unfilterable-float"` but I don't think that means we have to enforce that.

OTOH `multisampled:true,type:"float"` would be extra restrictive for no reason (wouldn't allow unfilterable float formats) so maybe it's best not to allow it.

I'm trying to make a table to understand this...